modules/datagrid/72/launch/added/launch/infinispan-config.sh

```diff
@@ -372,9 +372,22 @@ function configure_cache() {
                       <authorization $CACHE_SECURITY_AUTHORIZATION_ENABLED $CACHE_SECURITY_AUTHORIZATION_ROLES/>\
                     </security>"
   fi
-  if [ -n "$(find_env "${prefix}_CACHE_PARTITION_HANDLING_ENABLED")" ]; then
+  if [ -n "$(find_env "${prefix}_CACHE_PARTITION_HANDLING_ENABLED")$(find_env "${prefix}_CACHE_PARTITION_HANDLING_WHEN_SPLIT")$(find_env "${prefix}_CACHE_PARTITION_HANDLING_MERGE_POLICY")" ]; then
+
+    if [ -n "$(find_env "${prefix}_CACHE_PARTITION_HANDLING_ENABLED")" ]; then
+      log_warning "Ignoring paramater '${prefix}_CACHE_PARTITION_HANDLING_ENABLED' since 7.2, please use '${prefix}_CACHE_PARTITION_HANDLING_WHEN_SPLIT' and '${prefix}_CACHE_PARTITION_MERGE_POLICY' instead"
```

Comment:
Log says that parameter is ignored yet it configures <partition-handling /> element. No matter whether user had it set to 'false' or 'true'. We should stick up to generating same or similar configuration in new version even with old env var. Keeping the warning about deprecation of course.

Also take a look at [docs](https://access.redhat.com/documentation/en-us/red_hat_jboss_data_grid/7.2/html-single/administration_and_configuration_guide/#configuration_of_partition_handling_between_releases) which specify how to map old configuration to new one.

Aside that default values are according to [docs](https://access.redhat.com/documentation/en-us/red_hat_jboss_data_grid/7.2/html-single/administration_and_configuration_guide/#configure_partition_handling)
ALLOW_READ_WRITES when-split and PREFERRED_ALWAYS merge-policy. So it's quite probable that user that will migrate to new image stream will get quite different configuration then what he had.

Reply:
@Crumby The default values in the JDG docs are wrong, I changed the default merge-policy=NONE so that there would be less confusion for migrating users. Thanks for drawing this to my attention though, I will contact our docs writer.

I agree that if a user sets `_CACHE_PARTITION_HANDLING_ENABLED` as an env variable, we should set the config as before. The user will then get the internalised JDG warning as well as the config script warning. I'll add a `.feature` test for this new behaviour as well.

Reply:
@ryanemerson So I guess this ENV got to be completely broken in actual datagrid72 image. My suggestion to keep it convenient is:
* Print warning in case of deprecated env usage anyway
* If only deprecated env is used, set when-split and merge-policy based on [docs](https://access.redhat.com/documentation/en-us/red_hat_jboss_data_grid/7.2/html-single/administration_and_configuration_guide/#configuration_of_partition_handling_between_releases):
  * false -> <partition-handling when-split="ALLOW_READ_WRITES" merge-policy="NONE"/>
  * true -> <partition-handling when-split="DENY_READ_WRITES" merge-policy="NONE"/>
* If deprecated and new ENVs are used... Ignore old one.